src/popup/pages/TipsClaim.vue

```diff
@@ -70,83 +71,88 @@ export default {
   },
   setup(props, { root }) {
     const { activeAccount } = useAccounts({ store: root.$store });
+    const { openModal, openDefaultModal } = useModals();
+    const { getTippingContracts } = useTippingContracts({ store: root.$store });
 
-    return {
-      activeAccount,
-    };
-  },
-  data: () => ({
-    url: '',
-    loading: false,
-    BLOG_CLAIM_TIP_URL,
-  }),
-  computed: {
-    ...mapState(['tippingV1']),
-    ...mapGetters('sdkPlugin', ['sdk']),
-    ...mapGetters(['tippingSupported']),
-    normalizedUrl() {
-      if (!validateTipUrl(this.url)) return '';
-      return toURL(this.url).toString();
-    },
-  },
-  async mounted() {
-    if (IS_EXTENSION) {
-      const [tab] = await browser.tabs.query({ active: true, currentWindow: true });
-      if (tab && validateTipUrl(tab.url)) {
-        this.url = tab.url;
-      }
-    }
-  },
-  methods: {
-    async claimTips() {
-      const { openModal, openDefaultModal } = useModals();
-      const { activeAccount } = useAccounts({ store: this.$store });
-
-      const url = this.normalizedUrl;
-      this.loading = true;
-      await watchUntilTruthy(() => this.sdk && this.tippingV1);
+    const tipUrl = ref('');
+    const loading = ref(false);
+    const tippingSupported = computed(() => root.$store.getters.tippingSupported);
+
+    const normalizedUrl = computed(() => (
+      validateTipUrl(tipUrl.value)) ? toURL(tipUrl.value).toString() : '');
+
+    async function claimTips() {
+      const url = normalizedUrl.value;
+      loading.value = true;
       try {
+        const { tippingV1 } = await getTippingContracts();
         const claimAmount = parseFloat(
           aettosToAe(
-            await this.tippingV1.methods
-              .unclaimed_for_url(url)
+            await tippingV1.value.unclaimed_for_url(url)
               .then((r) => r.decodedResult)
               .catch(() => 1),
           ),
         );
         if (!claimAmount) {
           throw new Error('NO_ZERO_AMOUNT_PAYOUT');
         }
-        await this.$store.dispatch('claimTips', { url, address: activeAccount.value.address });
-        await this.$store.dispatch('cacheInvalidateOracle');
-        await this.$store.dispatch('cacheInvalidateTips');
+        await root.$store.dispatch('claimTips', { url, address: activeAccount.value.address });
+        await Promise.all([
+          root.$store.dispatch('cacheInvalidateOracle'),
+          root.$store.dispatch('cacheInvalidateTips'),
+        ]);
 
         openModal(MODAL_CLAIM_SUCCESS, { url, claimAmount });
 
-        this.$router.push({ name: ROUTE_ACCOUNT });
-      } catch (e) {
+        root.$router.push({ name: ROUTE_ACCOUNT });
+      } catch (e: any) {
```

Comment:
It would be awesome if you can also refactor the single character variable name to full `error` :)